lnwallet/channel.go

```diff
@@ -9999,3 +10081,17 @@ func (lc *LightningChannel) ZeroConfRealScid() fn.Option[lnwire.ShortChannelID]
 
 	return fn.None[lnwire.ShortChannelID]()
 }
+
+// entryTypeForHtlc returns the add type that should be used for adding this
+// HTLC to the channel. If the channel has a tapscript root and the HTLC carries
+// the NoOp bit in the custom records then we'll convert this to a NoOp add.
+func entryTypeForHtlc(records lnwire.CustomRecords,
```

Comment:
We need to expect `records` to be `nil`, since `Copy()` potentially returns `nil`.